src/sqlfluff/rules/L049.py

```diff
@@ -38,6 +38,9 @@ def _eval(self, context: RuleContext) -> LintResult:
         # Context/motivation for this rule:
         # https://news.ycombinator.com/item?id=28772289
         # https://stackoverflow.com/questions/9581745/sql-is-null-and-null
+        if not context.segment.is_type("expression"):
+            return LintResult()
```

Comment:
Added this as L049 can only apply to expression segments so no point doing the loops and various checks below for non expression segments 👍